src/misc/AttachShadowPolyfill.ts

```diff
@@ -0,0 +1,43 @@
+import { $$ } from '../utils/Dom';
+import 'styling/_AttachShadowPolyfill';
+
+export async function attachShadow(element: HTMLElement, init: ShadowRootInit): Promise<ShadowRoot | HTMLElement> {
+  if (element.attachShadow) {
+    return element.attachShadow(init);
+  }
+  const iframe = $$('iframe', { className: 'coveo-shadow-iframe', scrolling: 'no' }).el as HTMLIFrameElement;
+  const onLoad = new Promise(resolve => iframe.addEventListener('load', () => resolve()));
+  element.appendChild(iframe);
+  await onLoad;
+
+  const iframeBody = iframe.contentDocument.body as HTMLBodyElement;
+  iframeBody.style.margin = '0';
+  inheritStyle(iframeBody, document.body, ['fontFamily', 'fontSize', 'fontWeight']);
```

Comment:
The shadowDOM appears to inherit a few styles and looks better from it. Without this function, the font in the iframe is `Times New Roman` and looks too large.

Reply:
Would it be possible that we need to add more style to inherit depending on each client ?

Or would that list stays static like this ?

Reply:
It would be possible to add more styles to inherit, but it would probably be more practical for clients to redefine those styles in the embedded style.

Either way, since `iframe` is becoming the default, I'm going to remove style inheritance.